src/NHSD.BuyingCatalogue.Solutions.Persistence/Repositories/PriceRepository.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NHSD.BuyingCatalogue.Data.Infrastructure;
+using NHSD.BuyingCatalogue.Solutions.Contracts.Persistence;
+using NHSD.BuyingCatalogue.Solutions.Persistence.Models;
+
+namespace NHSD.BuyingCatalogue.Solutions.Persistence.Repositories
+{
+    public sealed class PriceRepository : IPriceRepository
+    {
+        private readonly IDbConnector _dbConnector;
+
+        private const string ListCataloguePricesSql = @"
+        SELECT  cp.CatalogueItemId,
+                ci.[Name] as CatalogueItemName,
+                cp.CataloguePriceId,
+                cp.CataloguePriceTypeId,
```

Comment:
Do we actually need the price type – won't `TieredPrice` be null when it's a flat price? (See my earlier comment too, on `PriceReader` I think.)